File: python/299.py
class Solution(object):
    def getHint(self, secret, guess):
        """
        :type secret: str
        :type guess: str
        :rtype: str
        """
        a = [0] * 10
        b = [0] * 10
        A = 0
        for i in range(len(guess)):
            if secret[i] == guess[i]:
                A += 1
            else:
                a[int(secret[i])] += 1
                b[int(guess[i])] += 1
        B = sum(min(a[i], b[i]) for i in range(10))
        return '{}A{}B'.format(A, B)

File: cpp/299.cpp
class Solution {
public:
  string getHint(string secret, string guess) {
    int a[10] = {0}, b[10] = {0};
    int A = 0, B = 0;
    for(int i=0; i<secret.size(); i++)
      if(secret[i] == guess[i])
        A++;
      else
        a[secret[i]-48]++, b[guess[i]-48]++;
    for(int i=0; i<10; i++)
      B += a[i] < b[i] ? a[i] : b[i];
    return to_string(A)+"A"+to_string(B)+"B";
  }
};
